storage/remote/generic/generic.go

```diff
@@ -30,9 +30,20 @@ type Client struct {
 }
 
 // NewClient creates a new Client.
-func NewClient(address string, timeout time.Duration) *Client {
-	// TODO: Do something with this error.
-	conn, err := grpc.Dial(address, grpc.WithInsecure(), grpc.WithTimeout(timeout))
+func NewClient(address string, compression string, timeout time.Duration) *Client {
+	var cp grpc.Compressor
+	switch compression {
+	case "gzip":
+		cp = grpc.NewGZIPCompressor()
+	case "snappy":
+		cp = &snappyCompressor{}
+	case "":
+		// No compression.
```

Comment:
It's funny the go-grpc only includes a built-in handler for gzip then and not the other compression types. I wonder why...